osu.Game.Rulesets.Mania/Skinning/Legacy/LegacyBodyPiece.cs

```diff
@@ -207,6 +210,29 @@ protected override void Update()
         {
             base.Update();
             missFadeTime.Value ??= holdNote.HoldBrokenTime;
+
+            // here we go...
+            switch (bodyStyle)
+            {
+                case LegacyNoteBodyStyle.Stretch:
+                    // this is how lazer works by default. nothing required.
+                    break;
+
+                default:
+                    // this is where things get fucked up.
+                    // honestly there's three modes to handle here but they seem really pointless?
+                    // let's wait to see if anyone actually uses them in skins.
+                    if (bodySprite != null)
+                    {
+                        var sprite = bodySprite as Sprite ?? bodySprite.ChildrenOfType<Sprite>().Single();
+
+                        bodySprite.FillMode = FillMode.Stretch;
+                        // i dunno this looks about right??
+                        bodySprite.Scale = new Vector2(1, 10000 / sprite.DrawHeight);
```

Comment:
Just to confirm - this is eyeballed, right? Not exact by any standard?

Reply:
Eyeballed, yep. I'll probably do a second pass if no one else does.

Reply:
FWIW I didn't manage to see how this compares against stable today. Could try tomorrow unless you beat me to it.

Reply:
Using the eggplant test skin, I've matched this to be as close to pixel perfect as I can (be bothered to).

| stable | lazer |
| -- | -- |
|![Parallels Desktop 2023-02-14 at 05 34 50](https://user-images.githubusercontent.com/191335/218648439-5a3c89cf-459e-4423-b355-702bc6df9818.png)|![osu! 2023-02-14 at 05 34 02](https://user-images.githubusercontent.com/191335/218648412-686ea7bd-8bee-42ee-80d3-17c0d9f310ab.png)|

Reply:
Tested N=4 skins with the latest changes. Seems close enough as far as I can afford to care. Let's just go with this and see what other curiosities (if any) people will pop up with...